HDWalletKit/Core/Crypto/Encryption/EllipticCurveEncrypterSecp256k1.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  EllipticCurveEncrypterSecp256k1.swift
+//  HDWalletKit
+//
+//  Created by Pavlo Boiko on 12.07.18.
+//  Copyright © 2018 Essentia. All rights reserved.
+//
+
+import secp256k1
+import CryptoSwift
+
+public class EllipticCurveEncrypterSecp256k1 {
+    // holds internal state of the c library
+    private let context: OpaquePointer
+    
+    public init() {
+        context = secp256k1_context_create(UInt32(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY))!
+    }
+    
+    deinit {
+        secp256k1_context_destroy(context)
+    }
+    
+    /// Signs the hash with the private key. Produces signature data structure that can be exported with
+    /// export(signature:) method.
+    ///
+    /// - Parameters:
+    ///   - hash: 32-byte (256-bit) hash of the message
+    ///   - privateKey: 32-byte private key
+    /// - Returns: signature data structure if signing succeeded, otherwise nil.
+    public func sign(hash: Data, privateKey: Data) -> secp256k1_ecdsa_recoverable_signature? {
+        precondition(hash.count == 32, "Hash must be 32 bytes size")
+        var signature = secp256k1_ecdsa_recoverable_signature()
+        let status = privateKey.withUnsafeBytes { (key: UnsafePointer<UInt8>) in
+            hash.withUnsafeBytes { secp256k1_ecdsa_sign_recoverable(context, &signature, $0, key, nil, nil) }
+        }
+        return status == 1 ? signature : nil
+    }
+    
+    /// Converts signature data structure to 65 bytes.
+    ///
+    /// - Parameter signature: signature data structure
+    /// - Returns: 65 byte exported signature data.
+    public func export(signature: inout secp256k1_ecdsa_recoverable_signature) -> Data {
+        var output = Data(count: 65)
+        var recid = 0 as Int32
```

Comment:
@impul what is recid?

Reply:
@kozlov-oleksandr, recid - a pointer to an integer to hold the recovery id

Reply:
@impul recId

Reply:
@kozlov-oleksandr done